packages/builder/src/registry.ts

```diff
@@ -700,13 +664,46 @@ export class OnChainRegistry extends CannonRegistry {
       account: this.signer.wallet.account || this.signer.address,
     };
 
-    txs.push(setPackageOwnershipParams);
-    txs.push(setAdditionalPublishersParams);
+    transactions.push(setPackageOwnershipParams);
+    transactions.push(setAdditionalPublishersParams);
 
-    const txData = prepareMulticall(txs);
+    return transactions;
+  }
+
+  prepareSequentialMulticall(transactions: any): TxData {
+    const nominatePackageOwnerTxs = [];
+    const setPackageOwnershipTxs = [];
+    const setAdditionalPublishersTxs = [];
+
+    for (const txn of transactions) {
+      switch (txn.functionName) {
+        case 'nominatePackageOwner': {
+          nominatePackageOwnerTxs.push(txn);
+          break;
+        }
+        case 'setPackageOwnership': {
+          setPackageOwnershipTxs.push(txn);
+          break;
+        }
+        case 'setAdditionalPublishers': {
+          setAdditionalPublishersTxs.push(txn);
+          break;
+        }
+        default:
+          throw new Error(`Unknown function name: ${txn.functionName}`);
+      }
+    }
+
+    return prepareMulticall([...nominatePackageOwnerTxs, ...setPackageOwnershipTxs, ...setAdditionalPublishersTxs]);
+  }
```

Comment:
Unless it is absolutely necessary, this function for reordering the calls is not necessary.